{"assignees":[],"author":{"id":"MDQ6VXNlcjI0NjI0MjEw","is_bot":false,"login":"aerdem4","name":"Ahmet Erdem"},"body":"**Describe the bug**\r\nI try to use groupby count transform for frequency encoding of a column. I get an error when I do it:\r\n\r\n> ValueError: Cannot align indices with non-unique values\r\n\r\n\r\n**Steps/Code to reproduce bug**\r\n`df[\"a_freq\"] = df.groupby([\"a\"])[\"b\"].transform(\"count\")\r\n`\r\nThis throws an error.\r\nSetting the column in advance with zero or using values work as workarounds.\r\n```\r\ndf[\"a_freq\"] = 0\r\ndf[\"a_freq\"] = df.groupby([\"a\"])[\"b\"].transform(\"count\")\r\n```\r\n`df[\"a_freq\"] = df.groupby([\"a\"])[\"b\"].transform(\"count\").values`\r\n\r\n**Expected behavior**\r\nI expect it to work as it works in pandas.\r\n\r\n**Environment overview (please complete the following information)**\r\n - Environment location: local\r\n - Method of cuDF install: conda\r\n\r\n**Environment details**\r\nconda environment with RAPIDS 22.02\r\n","closed":false,"closedAt":null,"comments":[{"id":"IC_kwDOBWUGps5C9-i2","author":{"login":"shwina"},"authorAssociation":"CONTRIBUTOR","body":"This looks like an issue that's potentially data dependant. Do you perhaps have a minimal reproducer of the problem?","createdAt":"2022-05-11T10:29:27Z","includesCreatedEdit":false,"isMinimized":false,"minimizedReason":"","reactionGroups":[],"url":"https://github.com/rapidsai/cudf/issues/10826#issuecomment-1123543222","viewerDidAuthor":false},{"id":"IC_kwDOBWUGps5EsRzg","author":{"login":"github-actions"},"authorAssociation":"NONE","body":"This issue has been labeled `inactive-30d` due to no recent activity in the past 30 days. Please close this issue if no further response or action is needed. Otherwise, please respond with a comment indicating any updates or changes to the original issue and/or confirm this issue still needs to be addressed. This issue will be labeled `inactive-90d` if there is no activity in the next 60 days.","createdAt":"2022-06-10T15:03:35Z","includesCreatedEdit":false,"isMinimized":false,"minimizedReason":"","reactionGroups":[],"url":"https://github.com/rapidsai/cudf/issues/10826#issuecomment-1152457952","viewerDidAuthor":false},{"id":"IC_kwDOBWUGps5J9upt","author":{"login":"github-actions"},"authorAssociation":"NONE","body":"This issue has been labeled `inactive-90d` due to no recent activity in the past 90 days. Please close this issue if no further response or action is needed. Otherwise, please respond with a comment indicating any updates or changes to the original issue and/or confirm this issue still needs to be addressed.","createdAt":"2022-09-08T16:03:08Z","includesCreatedEdit":false,"isMinimized":false,"minimizedReason":"","reactionGroups":[],"url":"https://github.com/rapidsai/cudf/issues/10826#issuecomment-1240918637","viewerDidAuthor":false}],"createdAt":"2022-05-11T10:01:53Z","id":"I_kwDOBWUGps5JdFtB","labels":[{"id":"MDU6TGFiZWw1OTk2MjY1NTk=","name":"bug","description":"Something isn't working","color":"d73a4a"},{"id":"MDU6TGFiZWwxMDEzOTg3Nzk5","name":"0 - Waiting on Author","description":"Waiting for author to respond to review","color":"ffb88c"},{"id":"MDU6TGFiZWwxMTM5NzQxMjEz","name":"cuDF (Python)","description":"Affects Python cuDF API.","color":"1d76db"}],"milestone":null,"number":10826,"projectCards":[{"project":{"name":"Bug Squashing"},"column":{"name":"Needs prioritizing"}}],"projectItems":[],"reactionGroups":[],"state":"OPEN","title":"[BUG] can't set groupby transform output to a new column","updatedAt":"2024-04-03T10:39:04Z","url":"https://github.com/rapidsai/cudf/issues/10826"}
